pytket/binders/circuit/Circuit/add_classical_op.cpp

```diff
@@ -237,20 +237,26 @@ void init_circuit_add_classical_op(
           "add_c_setreg",
           [](Circuit &circ, const _tket_uint_t value, const BitRegister &reg,
              const py::kwargs &kwargs) {
+	    unsigned int value_int = value;
+	    unsigned int tket_reg_width = _TKET_REG_WIDTH;
```

Comment:
Why are these conversions necessary? Also, this is the wrong check: we want to test against `reg.size()`, not `_TKET_REG_WIDTH`.

Reply:
> Why are these conversions necessary? 

If I don't do it, I get the following error:

```sh
 /Users/roland/Desktop/Projects/Tket/tket/pytket/binders/circuit/Circuit/add_classical_op.cpp:242:20: error: shift count >= width of type [-Werror,-Wshift-count-overflow]
                  if (value_int >> _TKET_REG_WIDTH != 0) {
                                ^  ~~~~~~~~~~~~~~~
      1 error generated.
```

which I interpret as being unable to define a right shift operator for an untyped variable defined in a macro.

> Also, this is the wrong check: we want to test against `reg.size()`, not `_TKET_REG_WIDTH`.

Is it ? My understanding was that we wanted to check if the value could be held on the tket register. But I understood that if I try to set a value that overflows the 32-bit register, I get a type error of some sort:

```sh
>>> a = c.add_c_register("a", 32)
>>> c.add_c_setreg(10 ** 24, a)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: add_c_setreg(): incompatible function arguments. The following argument types are supported:
    1. (self: pytket._tket.circuit.Circuit, value: int, arg: pytket._tket.circuit.BitRegister, **kwargs) -> pytket._tket.circuit.Circuit

Invoked with: [], 1000000000000000000000000, BitRegister("a", 32)
```

Unless you meant checking that a value could be held on a register of smaller size ?

Reply:
The value will always fit in 32 bits because it's a `uint32_t`. Don't we want to check that it fits in this particular register?

Reply:
Yes, a register of a smaller size. At first, I thought for a register of a larger size which didn't make much sense.

Reply:
Done.